scipy/spatial/tests/test_distance.py

```diff
@@ -258,12 +557,18 @@ def _check_calling_conventions(self, X1, X2, metric, eps=1e-07, **kwargs):
             if verbose > 2:
                 print(e_cls.__name__)
                 print(e)
-            assert_raises(e_cls, cdist, X1, X2, metric=metric, **kwargs)
-            assert_raises(e_cls, cdist, X1, X2, metric=eval(metric), **kwargs)
-            assert_raises(e_cls, cdist, X1, X2, metric="test_" + metric, **kwargs)
+            try:
+                assert_raises(Exception, cdist, X1, X2, metric=metric, **kwargs)
+                assert_raises(Exception, cdist, X1, X2, metric=eval(metric), **kwargs)
+                assert_raises(Exception, cdist, X1, X2, metric="test_" + metric, **kwargs)
+            except Exception as e2:
```

Comment:
Since this also tests some functions that don't support weights (mahalanobis, seuclidean, and sqeuclidean), that can't be done blindly. Maybe having a separate metric list for weighted would work for that though.